segmentstore/server/host/src/test/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessorTest.java

```diff
@@ -1724,6 +1728,85 @@ public void testTruncateIndexSegmentWithGreaterThenEndOffset() throws Exception
         assertEquals(0, store.getStreamSegmentInfo(indexSegment, PravegaRequestProcessor.TIMEOUT).join().getStartOffset());
     }
 
+    @Test
+    public void testTruncateIndexSegmentThrowStreamSegmentNotExistsException() throws Exception {
+        String segmentName = "truncateSegment";
+        String indexSegmentName = getIndexSegmentName(segmentName);
+        StreamSegmentStore store = mock(StreamSegmentStore.class);
+        @Cleanup
+        EmbeddedChannel channel = createChannel(store);
+
+        Duration timeout = Duration.ofMinutes(1);
+        ServiceBuilder serviceBuilder = newInlineExecutionInMemoryBuilder(getBuilderConfig());
+        serviceBuilder.initialize();
+
+        StreamSegmentInformation.builder()
+                .name(indexSegmentName)
+                .length(1000)
+                .startOffset(0)
+                .attributes(Map.of(EVENT_COUNT, 30L))
+                .build();
+
+        doThrow(new CompletionException(new StreamSegmentNotExistsException("Segment does not exits")))
+                .when(store).getStreamSegmentInfo(indexSegmentName, timeout);
+        doReturn(CompletableFuture.completedFuture(null)).when(store).truncateStreamSegment(anyString(), anyLong(), any());
+
+        Reply reply = sendRequest(channel, new WireCommands.TruncateSegment(requestId, segmentName, 30L, ""));
```

Comment:
Just above in doReturn, I am explicitly making truncateStreamSegment as success so that call can reach to the index segment truncation part.
It is well test and call is reaching to the index segment truncation part.